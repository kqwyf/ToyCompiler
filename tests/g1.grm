SS -> a
| ( L )
L -> L , SS
| SS
