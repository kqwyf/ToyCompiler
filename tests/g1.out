Symbol total: 8
State total: 13
Conflict total: 0

Symbols:

'S': 0
'#': 1
'SS': 2
'a': 3
'(': 4
'L': 5
')': 6
',': 7

FIRST:

S: ['a', '(']
#: ['#']
SS: ['a', '(']
a: ['a']
(: ['(']
L: ['a', '(']
): [')']
,: [',']

State:

I0:
S -> @ SS; #
SS -> @ a; #
SS -> @ ( L ); #

I1:
S -> SS @; #

I2:
SS -> a @; #

I3:
SS -> ( @ L ); #
L -> @ L , SS; )/,
L -> @ SS; )/,
SS -> @ a; )/,
SS -> @ ( L ); )/,

I4:
L -> L @ , SS; )/,
SS -> ( L @ ); #

I5:
L -> L , @ SS; )/,
SS -> @ a; )/,
SS -> @ ( L ); )/,

I6:
L -> L , SS @; )/,

I7:
SS -> a @; )/,

I8:
SS -> ( @ L ); )/,
L -> @ L , SS; )/,
L -> @ SS; )/,
SS -> @ a; )/,
SS -> @ ( L ); )/,

I9:
L -> L @ , SS; )/,
SS -> ( L @ ); )/,

I10:
SS -> ( L ) @; )/,

I11:
L -> SS @; )/,

I12:
SS -> ( L ) @; #


Table:

      #    a    (    )    ,    SS   L    
I0         s2   s3             1         
I1    acc                                
I2    r1                                 
I3         s7   s8             11   4    
I4                   s12  s5             
I5         s7   s8             6         
I6                   r3   r3             
I7                   r1   r1             
I8         s7   s8             11   9    
I9                   s10  s5             
I10                  r2   r2             
I11                  r4   r4             
I12   r2                                 

Conflicts:

